llvm/include/llvm/IR/Instruction.h

```diff
@@ -496,14 +496,23 @@ class Instruction : public User,
   /// Drops metadata that may generate poison.
   void dropPoisonGeneratingMetadata();
 
-  /// Return true if this instruction has poison-generating flags or metadata.
-  bool hasPoisonGeneratingFlagsOrMetadata() const {
-    return hasPoisonGeneratingFlags() || hasPoisonGeneratingMetadata();
+  /// Return true if this instruction has poison-generating attribute.
+  bool hasPoisonGeneratingAttribute() const LLVM_READONLY;
```

Comment:
```suggestion
  bool hasPoisonGeneratingReturnAttributes() const LLVM_READONLY;
```
Maybe? Just to make clear this won't report these attributes in argument position.

Reply:
yes think that it makes it much better have fixed it now